src/sentry/rules/conditions/tagged_event.py

```diff
@@ -49,50 +53,67 @@ def passes(self, event, state, **kwargs):
         match = self.get_option("match")
         value = self.get_option("value")
 
-        if not (key and match and value):
+        if not (key and match):
             return False
 
-        value = value.lower()
         key = key.lower()
+        tags = (k.lower() for k, v in event.tags)
```

Comment:
When we filter down the list of `values` below we use `k.lower() == key or tagstore.get_standardized_key(k) == key`. Since you also end up comparing these tags to `key` we probably need to do this as well here. How about we convert this to

`tags = set([k.lower() for k, v in event.tags] + [tagstore.get_standardized_key(k) for k, v in event.tags])`

You can then also change the logic in `IS_SET` to just be `return key in tags`, and `NOT_SET` to be `return key not in tags`

Reply:
I must say that my first approach was something similar to that, keeping the generator instead of generating a list. But then I thought that it's more optimal to iterate over the generator and returning as soon as it's found than consuming the whole thing just to use the membership operator.

But... I just checked and the membership operator is nicely implemented and it does exactly that, so it doesn't consume the whole generator if it finds the key in the middle. TIL

Back to your comment, you are suggesting to drop the generators and use a set made from the lists of the two tag sources? That seems suboptimal, as it will always have to iterate over all the tags. It most probably won't be a big issue (I guess), as there shouldn't be many tags? But I would keep the generator to be honest...

And you are right, I'm missing the `tagstore.get_standardized_key(k)` ones

Reply:
I don't mind too much about the `set` related things. You're right that it will do a little more work if we use them, so I'm happy to keep it as is and just compare both `tag` and `tagstore.get_standardized_key(tag)` in both `IS_SET` and `NOT_SET`

Reply:
Wait... do we really need that? I'm not familiar with Sentry's codebase, but that seems to be getting values from the key, which aren't necessary here 🤔

Reply:
Alright, so I changed it to 

```python
tags = (
    k
    for gen in (
        (k.lower() for k, v in event.tags),
        (tagstore.get_standardized_key(k) for k, v in event.tags),
    )
    for k in gen
)
```

As it will chain both operators, and I also changed it so it uses the membership operator, that I just found out that is as optimal as iterate over the generator and comparing